Index: installwatch/create-localdecls
===================================================================
--- installwatch/create-localdecls.orig
+++ installwatch/create-localdecls
@@ -64,37 +64,8 @@ if test "$VERSION" = 'libc.so.6' ; then
 	case "$OsLibcMajor" in
 	2)
 		# 2 is the glibc version
-		case "$OsLibcMinor" in
-		0)
-			echo '#define GLIBC_MINOR 0' >> $OUTFILE
-			SUBVERSION='glibc-2.0' ;;
-		1)
-			echo '#define GLIBC_MINOR 1' >> $OUTFILE
-			SUBVERSION='glibc-2.1' ;;
-		2)
-			echo '#define GLIBC_MINOR 2' >> $OUTFILE
-			SUBVERSION='glibc-2.2' ;;
-		3)
-			echo '#define GLIBC_MINOR 3' >> $OUTFILE
-			SUBVERSION='glibc-2.3' ;;
-		4)
-			echo '#define GLIBC_MINOR 4' >> $OUTFILE
-			SUBVERSION='glibc-2.4' ;;
-		5)
-			echo '#define GLIBC_MINOR 5' >> $OUTFILE
-			SUBVERSION='glibc-2.5' ;;
-		6)
-			echo '#define GLIBC_MINOR 6' >> $OUTFILE
-			SUBVERSION='glibc-2.6' ;;
-		7)
-			echo '#define GLIBC_MINOR 7' >> $OUTFILE
-			SUBVERSION='glibc-2.7' ;;
-		*)
-			echo 'Treated as glibc >= 2.1 (finger crossed)'
-			echo '#define GLIBC_MINOR 1' >> $OUTFILE
-			SUBVERSION='glibc-2.1' ;;
-	        esac
-		;;
+		echo '#define GLIBC_MINOR __GLIBC_MINOR__' >> $OUTFILE
+		SUBVERSION="glibc-2.$OsLibcMajor" ;;
 	esac
 fi
 
